src/core/jit/context.rs

```diff
@@ -8,21 +8,34 @@ use crate::core::ir::ResolverContextLike;
 pub struct Context<'a, Input, Output> {
     request: &'a Request<Input>,
     value: Option<&'a Output>,
+    is_query: bool,
 }
 
 impl<'a, Input, Output> Clone for Context<'a, Input, Output> {
     fn clone(&self) -> Self {
-        Self { request: self.request, value: self.value }
+        Self {
+            request: self.request,
+            value: self.value,
+            is_query: self.is_query,
+        }
     }
 }
 
 impl<'a, Input, Output> Context<'a, Input, Output> {
     pub fn new(request: &'a Request<Input>) -> Self {
-        Self { request, value: None }
+        Self { request, value: None, is_query: false }
```

Comment:
I suggest to pass `is_query` as arguments to `new` instead of separate `with_is_query` since it's significant part of the processed query and it'll be harder to forget about it